File: covered-pawns/covered-pawns.py
def covered_pawns(pawns):
    safe = len(pawns)
    for i in pawns:
        if not (set.intersection(area(i),set(pawns))):
            safe -=1
    return safe
    
def area(i):
    st  = "0abcdefgh0"    
    a = st[st.index(i[0])-1]+str(int(i[1])-1)
    b = st[st.index(i[0])+1]+str(int(i[1])-1)
    return {a,b}

File: covered-pawns/covered-pawns.js
function coveredPawns(pawns){
    var st  = "0abcdefgh0"
    return pawns.filter(a=>pawns.indexOf(st[st.indexOf(a[0])-1]+((a[1])-1))!=-1||pawns.indexOf(st[st.indexOf(a[0])+1]+((a[1])-1))!=-1).length;}